sandbox/modules/identity/src/main/java/org/opensearch/identity/rest/user/UserService.java

```diff
@@ -232,6 +236,76 @@ protected void saveAndUpdateConfiguration(
         }
     }
 
+    public void resetPassword(ResetPasswordRequest request, ActionListener<ResetPasswordResponse> listener) {
+        if (!ensureIndexExists()) {
+            listener.onFailure(new IndexNotFoundException(ErrorType.IDENTITY_NOT_INITIALIZED.getMessage()));
+            return;
+        }
+
+        String username = request.getUsername();
+        String oldPassword = request.getOldPassword();
+        String newPassword = request.getNewPassword();
+        String newPasswordValidation = request.getNewPasswordValidation();
```

Comment:
newpasswordverify is a good idea to ensure the user doesn't make a typo and is unable to login to use their account after a password reset, but this functionality may make more sense on the client side than in here. The security plugin's equivalent API for this action only takes in the new password once.

Reply:
Ya I have removed this.